internal/common/conversion/diags.go

```diff
@@ -34,13 +49,3 @@ func AddJSONBodyErrorToDiagnostics(msgPrefix string, err error, diags *diag.Diag
 	errorJSON := string(errorBytes)
 	diags.AddError(msgPrefix, errorJSON)
 }
-
-func AddLegacyDiags(diags *diag.Diagnostics, legacyDiags legacyDiag.Diagnostics) {
```

Comment:
@EspenAlbert noticed function was not being used so prefer to remove. Let me know if you do consider preserving it.